Using Mesh Term as My Repors predicate throws error

building a report along the lines of "get all XXX associated with the MeSH term ZZZ" throws the following error:
```
ilios-07af94bbfd8694db87aad03f5c42ab08.js:2843 Error: a glimmer transaction was begun, but one already exists. You may have a nested transaction
    at e.assert (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2356)
    at i.e.begin (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2264)
    at i.begin (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2825)
    at n.e._renderRoots (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2956)
    at n.e._renderRootsTransaction (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2961)
    at n.e._revalidate (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2963)
    at e.invokeWithOnError (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2390)
    at e.flush (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2377)
    at e.flush (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2393)
    at e.end (vendor-f2e927e04b8abbbc4d500c718a23a4e3.js:2401)
```

hard refresh is required to clear the form and move forward.

--- a/app/components/mesh-manager.js
+++ b/app/components/mesh-manager.js
@@ -2,12 +2,17 @@ import { inject as service } from '@ember/service';
 import ObjectProxy from '@ember/object/proxy';
 import Component from '@ember/component';
 import { computed } from '@ember/object';
+import { isEmpty } from '@ember/utils';
 import { task } from 'ember-concurrency';
 import layout from '../templates/components/mesh-manager';
 
-var ProxiedDescriptors = ObjectProxy.extend({
+const ProxiedDescriptors = ObjectProxy.extend({
   terms: null,
-  isActive: computed('content', 'terms.[]', function(){
+  isActive: computed('content', 'terms.[]', function () {
+    const terms = this.get('terms');
+    if (isEmpty(terms)) {
+      return true;
+    }
     return !this.get('terms').includes(this.get('content'));
   })
 });
@@ -20,6 +25,7 @@ export default Component.extend({
     this.set('searchResults', []);
     this.set('sortTerms', ['name']);
   },
+  'data-test-mesh-manager': true,
   layout: layout,
   classNames: ['detail-block', 'mesh-manager'],
   terms: null,

--- a/app/components/search-box.js
+++ b/app/components/search-box.js
@@ -8,6 +8,7 @@ export default Component.extend({
   classNames: ['search-box'],
   value: '',
   liveSearch: true,
+  'data-test-search-box': true,
   searchTask: task(function * () {
     yield timeout(DEBOUNCE_TIMEOUT);
     const value = this.get('value');

--- a/app/templates/components/mesh-manager.hbs
+++ b/app/templates/components/mesh-manager.hbs
@@ -32,10 +32,10 @@
 {{else}}
   {{#if searchResults.length}}
     <div class='selectable-list'>
-      <ul class='mesh-search-results'>
+      <ul class='mesh-search-results' data-test-search-results>
         {{#each searchResults as |term|}}
           <li class="{{unless term.isActive 'disabled'}}" {{action 'add' term}}>
-            <span class="descriptor-name">{{term.name}}</span>
+            <span class="descriptor-name" data-test-name>{{term.name}}</span>
             <span class="descriptor-id">
               {{term.id}}
               {{#if term.trees }}

--- a/tests/acceptance/dashboard/reports-test.js
+++ b/tests/acceptance/dashboard/reports-test.js
@@ -174,3 +174,31 @@ test('filter session by year in new report form', async function (assert) {
   assert.equal(page.myReports.selectedReport.results().count, 1);
   assert.equal(page.myReports.selectedReport.results(0).text, 'Vocabulary 1 > term 0');
 });
+
+test('get all courses associated with mesh term #3419', async function (assert) {
+  server.create('mesh-descriptor', {
+    id: 'D1234',
+    courseIds: [1, 2]
+  });
+  await page.visit();
+  assert.equal(page.myReports.reports().count, 2);
+  await page.myReports.addNewReport();
+
+  await page.myReports.newReport.chooseSchool('school 0');
+  await page.myReports.newReport.chooseSubject('Courses');
+  await page.myReports.newReport.chooseObjectType('MeSH Term');
+  await page.myReports.newReport.fillMeshSearch('0');
+  await page.myReports.newReport.runMeshSearch();
+  assert.equal(page.myReports.newReport.meshSearchResults().count, 1);
+  assert.equal(page.myReports.newReport.meshSearchResults(0).text, 'descriptor 0 D1234');
+  await page.myReports.newReport.meshSearchResults(0).pick();
+  await page.myReports.newReport.save();
+
+  assert.equal(page.myReports.reports().count, 3);
+  await page.myReports.reports(1).select();
+
+  assert.equal(page.myReports.selectedReport.title, 'All Courses for descriptor 0 in school 0');
+  assert.equal(page.myReports.selectedReport.results().count, 2);
+  assert.equal(page.myReports.selectedReport.results(0).text, '2015 - 2016 course 0');
+  assert.equal(page.myReports.selectedReport.results(1).text, '2016 - 2017 course 1');
+});

--- a/tests/pages/dashboard.js
+++ b/tests/pages/dashboard.js
@@ -45,6 +45,15 @@ export default create({
       chooseObject: selectable('[data-test-report-object]'),
       objectCount: count('[data-test-report-object] option'),
       chooseAcademicYear: selectable('[data-test-report-year-filter]'),
+      fillMeshSearch: fillable('[data-test-mesh-manager] [data-test-search-box] input'),
+      runMeshSearch: clickable('[data-test-mesh-manager] [data-test-search-box] .search-icon'),
+      meshSearchResults: collection({
+        itemScope: '[data-test-search-results] li',
+        item: {
+          name: text('[data-test-name]'),
+          pick: clickable()
+        },
+      }),
       save: clickable('[data-test-report-save]'),
     }
   },

